partner_firstname/views/res_partner.xml

```diff
@@ -10,20 +10,20 @@
         <data>
             <xpath expr="//field[@name='name']" position="attributes">
                 <attribute name="attrs">{
-                    'readonly': [('is_company', '=', False)],
-                    'required': [('is_company', '=', True)]
+                    'readonly': [('company_type', '=', 'person')],
```

Comment:
company_type is only a technical field to have the radio button selection. This is not needed.